nexus/src/app/instance_platform/cpu_platform.rs

```diff
@@ -0,0 +1,1193 @@
+            // target has fp128, that's probably fine. Likewise, if the base has
+            // fp512 and the target has a smaller width, claiming the FPU
+            // datapath is narrower than it really is, is probably fine.
+            if base_info.has_fp256() != target_info.has_fp256() {
+                return false;
+            }
+        }
+        _ => {
+            // Specific cases here may be acceptable, but for expediency (and
+            // because we don't intend to support vCPUs whose profiles would not
+            // have this leaf), just bail here.
+            return false;
+        }
+    }
+
+    // Bits checked here describe architectural behavior. If they differ, the
+    // base CPU will behave differently than the target wants to see.
+    //
+    // It may be okay in some cases to allow these to differ, but take a
+    // conservative approach until we need otherwise.
+    match (
+        base.get_extended_feature_identification_2(),
+        target.get_extended_feature_identification_2(),
+    ) {
+        (Some(base_info), Some(target_info)) => {
+            if base_info.has_no_nested_data_bp()
+                != target_info.has_no_nested_data_bp()
+            {
+                return false;
+            }
+
+            if base_info.has_lfence_always_serializing()
+                != target_info.has_lfence_always_serializing()
+            {
+                return false;
+            }
+
+            if base_info.has_null_select_clears_base()
+                != target_info.has_null_select_clears_base()
+            {
+                return false;
+            }
+        }
+        _ => {
+            return false;
+        }
+    }
+
+    // TODO: really not sure if we should include things like cache
+    // hierarchy/core topology information here. Misrepresenting the actual
+    // system can result in cache-sized buffers being sized incorrectly (or at
+    // least suboptimally). If the processor caches do not match the reported
+    // topology, software that tries to use the topology to flush caches by
+    // flushing each set individually would not behave correctly; if caches are
+    // larger than reported, such a manual flushing procedure would not actually
+    // flush the whole cache. Anyone doing this *really* should use WBINVD (or
+    // INVD) anyway.
+
+    true
+}
+
+/// The Platonic ideal Milan. This is what we would "like" to define as "The
+/// Milan vCPU platform" absent any other constraints. This is a slightly
+/// slimmer version of the Milan platform defined in RFD 314, with
+/// justifications there.
+///
+/// Notably, this avoids describing individual processor SKUs' characteristics,
+/// where possible.  This CPUID configuration as-is is untested; guests may not
+/// boot, this may be too reductive, etc.
+fn milan_ideal() -> CpuIdDump {
+    let mut cpuid = CpuId::with_cpuid_reader(CpuIdDump::new());
```

Comment:
because this whole function is just constructing a pile of bits, it really feels like it _should_ be able to be done in const-eval, but it looks like the raw-cpuid API won't let you do that. ah well.

Reply:
yeah, i think it'd be reasonable to flip this and have the const arrays be the source of truth as-is. but it feels like we're closer than not to needing to poke a bit here and there based on some VM configuration so that'd have limited utility.

on the upside, i think most of the bit twiddling here gets mushed together when optimized, so this is something like 20 allocs, ~800 stores, and maybe 30 map inserts? i haven't really looked, though.